src/reporter.ts

```diff
@@ -53,22 +53,35 @@ export const shouldTrim = (spans, maxSendBytes: number): boolean => {
 };
 
 const isJsonContent = (payload: any, headers: Object) => {
-  return isString(payload) && headers['content-type'] && headers['content-type'].includes('json');
+  return isString(payload) && headers['content-type']?.includes('json');
 };
 
-function scrub(payload: any, headers: any, sizeLimit: number, truncated = false): string {
+function scrub(
+  payload: any,
+  headers: any,
+  scrubContext: ScrubContext.HTTP_REQUEST_BODY | ScrubContext.HTTP_RESPONSE_BODY,
+  sizeLimit: number,
+  truncated = false
+): string {
   try {
     if (isJsonContent(payload, headers)) {
-      if (truncated) payload = untruncateJson(payload);
-      return payloadStringify(JSON.parse(payload), sizeLimit, null, truncated);
-    } else {
-      return payloadStringify(payload, sizeLimit, truncated);
+      const jsonPayload = truncated ? untruncateJson(payload) : payload;
+      return payloadStringify(JSON.parse(jsonPayload), scrubContext, sizeLimit, null, truncated);
     }
   } catch (e) {
-    return payloadStringify(payload, sizeLimit, truncated);
+    logger.warn('An error occurred while stringifying JSON payload', e);
   }
+
+  return payloadStringify(payload, scrubContext, sizeLimit, null, truncated);
 }
 
+/*
+ * TODO! We should _not_ scrub spans *again*, this already done (or should be done) in the
```

Comment:
please remove the TODOs and create tickets to do the work instead

Reply:
I am against this. JIRA is where important items like these go to die. I want each and every one of us looking at this code go "DAMN we should do that" :D @dori-admin wdyt.